internal/instance/resource_instance_test.go

```diff
@@ -216,6 +216,55 @@ func TestAccInstance_restartVirtualMachine(t *testing.T) {
 	})
 }
 
+func TestAccInstance_renameInstance(t *testing.T) {
+	instanceNameA := acctest.GenerateName(3, "-")
+	instanceNameB := acctest.GenerateName(3, "-")
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:                 func() { acctest.PreCheck(t) },
+		ProtoV6ProviderFactories: acctest.ProtoV6ProviderFactories,
+		Steps: []resource.TestStep{
+			{
+				// Launch a new instance.
+				Config: testAccInstance_rename(instanceNameA, true),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("lxd_instance.instance1", "name", instanceNameA),
+					resource.TestCheckResourceAttr("lxd_instance.instance1", "status", "Running"),
+				),
+			},
+			{
+				// Stop and rename the instance.
+				Config: testAccInstance_rename(instanceNameB, false),
```

Comment:
The provided name could be a bit more indicative of the intention. Say the instance started with `A`, then move to `A2B` -> `B2C` -> `C2D` -(fail)-> `D2E`.

Just an idea as I think what you have is fine, it just required a bit more mental tracking.

Reply:
Hehe, will fix :)